assets/js/components/KeyMetrics/ConnectGA4CTAWidget.js

```diff
@@ -0,0 +1,208 @@
+import { SpinnerButton } from 'googlesitekit-components';
+import KeyMetricsCTAContent from './KeyMetricsCTAContent';
+import KeyMetricsCTAFooter from './KeyMetricsCTAFooter';
+import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import { CORE_WIDGETS } from '../../googlesitekit/widgets/datastore/constants';
+import { AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY } from '../../googlesitekit/widgets/default-areas';
+import {
+	FORM_SETUP,
+	MODULES_ANALYTICS,
+} from '../../modules/analytics/datastore/constants';
+import { CORE_LOCATION } from '../../googlesitekit/datastore/location/constants';
+import { CORE_FORMS } from '../../googlesitekit/datastore/forms/constants';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import useActivateModuleCallback from '../../hooks/useActivateModuleCallback';
+import useCompleteModuleActivationCallback from '../../hooks/useCompleteModuleActivationCallback';
+import { useDebounce } from '../../hooks/useDebounce';
+import { snapshotAllStores } from '../../googlesitekit/data/create-snapshot-store';
+const { useSelect, useDispatch } = Data;
+
+export default function ConnectGA4CTAWidget( { Widget, WidgetNull } ) {
+	const DISMISSED_ITEM_KEY = 'key-metrics-connect-ga4-cta-widget';
+
+	const isCTADismissed = useSelect( ( select ) =>
+		select( CORE_USER ).isItemDismissed( DISMISSED_ITEM_KEY )
+	);
+	const isUserInputCompleted = useSelect( ( select ) =>
+		select( CORE_USER ).isUserInputCompleted()
+	);
+	const isGA4Connected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected( 'analytics-4' )
+	);
+	const ga4DependantKeyMetrics = useSelect( ( select ) => {
+		const keyMetrics = select( CORE_USER ).getKeyMetrics();
+		const widgets = select( CORE_WIDGETS ).getWidgets(
+			AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY
+		);
+
+		if ( ! keyMetrics || ! widgets ) {
+			return [];
+		}
+
+		return widgets.filter(
+			( { slug, modules } ) =>
+				keyMetrics.includes( slug ) && modules.includes( 'analytics-4' )
+		);
+	} );
+	const isAnalyticsActive = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleActive( 'analytics' )
+	);
+	const isNavigatingToReauthURL = useSelect( ( select ) => {
+		const adminReauthURL = select( MODULES_ANALYTICS ).getAdminReauthURL();
+
+		if ( ! adminReauthURL ) {
+			return false;
+		}
+
+		return select( CORE_LOCATION ).isNavigatingTo( adminReauthURL );
+	} );
+	const isActivatingAnalytics = useSelect( ( select ) =>
+		select( CORE_MODULES ).isFetchingSetModuleActivation(
+			'analytics',
+			true
+		)
+	);
+	const settingsURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getAdminURL( 'googlesitekit-settings' )
+	);
+	const isAnalyticsConnected = useSelect( ( select ) =>
+		select( CORE_MODULES ).isModuleConnected( 'analytics' )
+	);
+	const connectGA4URL = `${ settingsURL }#connected-services/analytics/edit`;
```

Comment:
This can be combined into the selector for `settingsURL` since that isn't used anywhere else 👍

Reply:
Updated, thanks!

Reply:
Updated, thanks!